### PR TITLE
Game model with methods

### PR DESCRIPTION
**Pull request by MichalKalke**

### CODE REVIEW

File: internal/model/game.go [hunk 1]
```diff
@@ -1 +1,20 @@
 package model
+
+type GameScore struct {
+	BlueScore  int `json:"blueScore"`
+	WhiteScore int `json:"whiteScore"`
+}
```
Comment (pPrecel) [inline internal/model/game.go:6]: Is there a sense to keeping a public struct? Maybe better would be to add a method that will return the pointer to the private method? 

```suggestion
type gameScore struct {
	BlueScore  int `json:"blueScore"`
	WhiteScore int `json:"whiteScore"`
}

func New() *gameScore {
	return &gameScore{}
}
```

It prevents misunderstanding related to importing packages when a user would not know if he should create a struct manually or use a specific function

File: internal/model/game_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package model
```
Comment (pPrecel) [inline internal/model/game_test.go:1]: Good job!

**Review by pPrecel — APPROVED** [review]
LGTM!

One suggestion only

File: internal/model/game.go [hunk 3]
```diff
@@ -1 +1,24 @@
...
+
+func (gameScore *GameScore) AddGoal(teamID int) {
+	switch teamID {
+	case 1:
+		gameScore.WhiteScore++
+	case 2:
+		gameScore.BlueScore++
+	}
+}
```
Comment (Pichi00) [inline internal/model/game.go:24]:
```suggestion
func (gameScore *gameScore) ResetScore() {
	gameScore.BlueScore = 0
	gameScore.WhiteScore = 0
}

func (gameScore *gameScore) AddGoal(teamID int) {
	switch teamID {
	case 1:
		gameScore.WhiteScore++
	case 2:
		gameScore.BlueScore++
	}
}
```

File: internal/model/game_test.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResetScore(t *testing.T) {
+	gameScore := &GameScore{3, 1}
+	type args struct {
+		score *GameScore
+	}
```
Comment (Pichi00) [inline internal/model/game_test.go:13]:
```suggestion
	gameScore := &gameScore{3, 1}
	type args struct {
		score *gameScore
	}
```

File: internal/model/game_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+			gameScore.ResetScore()
+		})
+		if tt.args.score.WhiteScore != 0 || tt.args.score.BlueScore != 0 {
+			t.Errorf("Score did not reset. Goals white: %v, Goals blue: %v", tt.args.score.WhiteScore, tt.args.score.BlueScore)
+		}
+	}
+}
+
+func TestAddGoal(t *testing.T) {
+	gameScore := &GameScore{3, 1}
```
Comment (Pichi00) [inline internal/model/game_test.go:31]:
```suggestion
	gameScore := &gameScore{3, 1}
```

File: internal/model/game_test.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+		})
+		if tt.args.score.WhiteScore != 0 || tt.args.score.BlueScore != 0 {
+			t.Errorf("Score did not reset. Goals white: %v, Goals blue: %v", tt.args.score.WhiteScore, tt.args.score.BlueScore)
+		}
+	}
+}
+
+func TestAddGoal(t *testing.T) {
+	gameScore := &gameScore{3, 1}
+	type args struct {
```
Comment (VOID404) [inline internal/model/game_test.go:32]: Does this struct improve the tests in any way?

Comment (VOID404) [inline internal/model/game_test.go:32]: what's the benefit of adding this struct?

File: internal/model/game_test.go [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+		team int
+	}
+	tests := []struct {
+		name             string
+		args             args
+		wantedBlueScore  int
+		wantedWhiteScore int
```
Comment (VOID404) [inline internal/model/game_test.go:39]: `wanted` -> `expected`

File: internal/model/game_test.go [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+	}{
+		{"should increment team white score by one", args{1}, 0, 1},
+		{"should increment team blue score by one", args{2}, 1, 0},
+		{"should not cause an error", args{-1}, 0, 0},
```
Comment (VOID404) [inline internal/model/game_test.go:43]: Wouldn't this be more readable with explicit field names?

Comment (VOID404) [inline internal/model/game_test.go:43]: Shouldn't an incorrect team cause error?

**Pull request merged**